### PR TITLE
fix(ui): order outcomes by reaction time in the reaction view (#599)

### LINKED ISSUE #599: Ordering of outcomes by reaction time does not work
It's from AC#4 https://github.com/open-reaction-database/ord-app/issues/53:
All Outcomes should be demonstrated according to the reaction time order if it's available, otherwise they should be displayed in the same order as they are stored.

## Steps to Reproduce 
1.  Add a reaction with several outcomes and with different reaction time:
Image
2. Observe the preview
 
## Actual Result 
-  Ordering by time doesn't work:
Image

## Expected Result 
-  All mentioned parameters should be displayed on preview (the second product should be the first)

## Environment 
- Reproduced in the local environment with main branch, commit   e6b3d7adeb58ebc2dcd481cc2883d7c636af07ba;
- dev-z4acb31kcl4prqtw.us.auth0.com auth tenant;
- Desktop, Windows 11 OS, Chrome Browser.

## Note
- It also doesn't work in the original app.

### PR DESCRIPTION
**Pull request by skearnes**
## Summary

Per **AC#4 of #53** (#599): outcomes should display **ordered by reaction time when available, otherwise in stored order**. The Outcomes list rendered them in raw stored order.

### Fix
New `sortOutcomesByReactionTime` helper:
- Normalizes reaction time to **seconds** (`DAY`/`HOUR`/`MINUTE`/`SECOND`) so outcomes recorded in different units sort correctly.
- **Stable** ascending sort; **non-mutating**.
- Preserves each outcome's **original stored index**, which is passed as `outcomeIndex` (the edit path is `['outcomes', index]`) — so reordering the display never targets the wrong outcome when editing. `outcomeIndex` is path-only (not a displayed number), so there's no numbering side effect.

### ⚠️ Interpretation to confirm (review)
The AC is explicit for the all-timed case but underspecified when **some** outcomes lack a reaction time. I chose: **timed outcomes first (ascending), then untimed in stored order** (an unspecified/unknown unit counts as "no usable time"). If you'd rather untimed outcomes stay interleaved in stored position, or only reorder when *every* outcome has a time, say so and I'll adjust.

## Test plan
- [x] `npx tsc -b` clean
- [x] `npx vitest run` — 6 new `sortOutcomesByReactionTime` tests (ascending order + index preserved; cross-unit normalization; untimed kept in stored order after timed; all-untimed unchanged; unspecified unit treated as untimed; input not mutated). Full suite green.
- [x] `prettier` / `eslint` clean
- [ ] Held for review — behavioral display change; the runtime setup (multiple outcomes with differing reaction times) is awkward to script, and the sort logic is fully unit-covered.

Closes #599.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h3>Greptile Summary</h3>

Introduces `sortOutcomesByReactionTime` to display reaction outcomes in ascending reaction-time order, falling back to stored order for untimed outcomes, and wires it into `Outcomes.tsx` while preserving original stored indices for correct edit paths.

- New `reactionOutcomes.utils.ts` normalises reaction time to seconds, uses a `satisfies`-enforced map so any future schema unit addition breaks the build rather than silently misbehaving, and performs a stable non-mutating sort.
- `Outcomes.tsx` adds a single `useMemo` for the sorted list and destructures `{ outcome, index }` so `outcomeIndex` passed to `OutcomeListItem` always reflects the stored position, not the display position.
- Six unit tests cover the key behaviours: ascending order, cross-unit normalisation, untimed-after-timed, all-untimed stability, UNSPECIFIED-unit treatment, and input immutability.

<h3>Confidence Score: 5/5</h3>

Safe to merge — isolated display-order change with no mutation of stored data and full preservation of edit paths.

The change is narrowly scoped to rendering order. The sort helper is non-mutating, original stored indices flow through unchanged to `outcomeIndex`, and the unit-exhaustiveness constraint is compile-time enforced. The six unit tests cover every documented edge case. No store writes, selectors, or API calls are touched.

No files require special attention.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| ui/src/store/entities/reactions/reactionsOutcomes/reactionOutcomes.utils.ts | New utility: `reactionTimeInSeconds` normalizer + `sortOutcomesByReactionTime` exported function. Compile-time exhaustive unit map via `satisfies`, stable non-mutating sort, preserves original stored index. |
| ui/src/store/entities/reactions/reactionsOutcomes/reactionOutcomes.utils.test.ts | Six focused unit tests covering ascending sort, cross-unit normalization, untimed-after-timed ordering, all-untimed stability, UNSPECIFIED unit treatment, and non-mutation. Good coverage of the defined behaviour. |
| ui/src/features/reactions/ReactionView/Outcomes/Outcomes.tsx | Adds `orderedOutcomes` memo via `sortOutcomesByReactionTime`, replaces raw `.map` with destructured `{ outcome, index }` iteration, passes original stored `index` as `outcomeIndex` to preserve edit paths. Minimal and correct change. |

</details>

</details>

<sub>Reviews (2): Last reviewed commit: ["refactor(ui): make the outcome time-unit..."](https://github.com/open-reaction-database/ord-app/commit/efefbeab53d918a8dff6d5408677e96709c7a4d8) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=37453074)</sub>

### CODE REVIEW

File: ui/src/store/entities/reactions/reactionsOutcomes/reactionOutcomes.utils.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import type { ReactionOutcome } from './reactionOutcomes.types.ts';
+import type { ReactionTime } from '../reactionEntity/reactionEntity.types.ts';
+
+// Reaction-time units normalized to seconds so outcomes recorded in different units sort correctly.
+const TIME_UNIT_TO_SECONDS: Record<string, number> = {
+  DAY: 86_400,
+  HOUR: 3_600,
+  MINUTE: 60,
+  SECOND: 1,
+};
```
[the remark this answers is not in the document]
Comment (skearnes) [reply ui/src/store/entities/reactions/reactionsOutcomes/reactionOutcomes.utils.ts]: Good future-proofing point. The current `ord.Time.TimeUnit` is exactly `UNSPECIFIED/DAY/HOUR/MINUTE/SECOND` (no WEEK), so the map was complete — but I made exhaustiveness **compile-time enforced**: the map now uses `satisfies Record<Exclude<ReactionTimeType, 'UNSPECIFIED'>, number>`, so adding any unit to the schema breaks the build until it gets a seconds factor (rather than silently sorting as untimed).

🤖 Generated with [Claude Code](https://claude.com/claude-code)
